app/routes.py

```diff
@@ -1,2 +1,293 @@
+    video = Video.query.get(video_id)
+    form_data = request.get_json()
+
+    if video == None:
+        return jsonify({"errors": "Video not found"}), 404
+    if form_data == {}:
+        return jsonify(INVALID_DATA_ERROR), 400
+
+    video.title = form_data['title']
+    video.release_date = form_data['release_date']
+    video.total_inventory = form_data['total_inventory']
+    db.session.commit()
+
+    return jsonify(video.vid_details()), 200
+
+@video_bp.route("/<video_id>", methods=["DELETE"])
+def delete_video(video_id):
+    video = Video.query.get(video_id)
+
+    if video == None:
+        return jsonify({"errors": "Video not found"}), 404
+
+    db.session.delete(video)
+    db.session.commit()
+
+    return jsonify({"id": video.video_id}), 200
+
+@rental_bp.route("/check-out", methods = ["POST"])
+def check_out_video():
+    request_body = request.get_json()
+    keys = ["video_id", "customer_id"]
+    if dict_helper.missing_any(request_body, keys):
+        return jsonify(INVALID_DATA_ERROR), 400
+    if not isinstance(request_body["video_id"], int) or not isinstance(request_body["customer_id"], int):
+        return jsonify(INVALID_DATA_ERROR), 400
+
+    vid_id = request_body["video_id"]
+    cust_id = request_body["customer_id"]
+    video = Video.query.get(vid_id)
+    customer = Customer.query.get(cust_id)
+
+    if video == None or customer == None:
+        return jsonify(INVALID_DATA_ERROR), 404
+    if video.available_inventory == 0:
+        return jsonify({"error": "Out of stock."}), 400
+
+    video.check_out()
+    customer.check_out()
+
+    new_rental = Rental(
+        video_id = request_body["video_id"],
+        customer_id = request_body["customer_id"],
+        due_date = datetime.datetime.now()+datetime.timedelta(days=7)
+    )
+    db.session.add(new_rental)
+    db.session.commit()
+
+    return jsonify({
+        "customer_id": new_rental.customer_id,
+        "video_id": new_rental.video_id,
+        "due_date": new_rental.due_date,
+        "videos_checked_out_count": customer.videos_checked_out_count,
+        "available_inventory": video.available_inventory
+        }), 200
+
+@rental_bp.route("/check-in", methods = ["POST"])
+def check_in_video():
+    request_body = request.get_json()
+    vid_id = request_body["video_id"]
+    cust_id = request_body["customer_id"]
+    video = Video.query.get(vid_id)
+    customer = Customer.query.get(cust_id)
+    check_if_rental = Rental.query.filter_by(video_id=vid_id, customer_id=cust_id).all()
+
+    if "video_id" not in request_body or "customer_id" not in request_body:
+        return jsonify(INVALID_DATA_ERROR), 400
+    if not isinstance(request_body["video_id"], int) or not isinstance(request_body["customer_id"], int):
+        return jsonify(INVALID_DATA_ERROR), 400
+    if video == None or customer == None:
+        return jsonify(INVALID_DATA_ERROR), 404
+    if len(check_if_rental) == 0:
+        return jsonify(INVALID_DATA_ERROR), 400
```

Comment:
The code is so DRY in general! This is one area I notice where there could probably be a nice helper function that could be called by check_in and check_out, since its a lot of the same validation on the same two params.